### DISCUSSION TITLE
Reacting on any keypress

### DISCUSSION THREAD

**Discussion by smallB007** [gyscos/cursive#713] [General]
Hi
How to create Dialog that would react to any keypress. I know I can wrap it in OnEventView but then I literally have to list every char and attach .on_event on every char. Is there an easier way?
Thank you

**Comment by gyscos** [comment]
Hi!

With `OnEventView` you can register any [`T: Into<EventTrigger>`](https://docs.rs/cursive/latest/cursive/views/struct.OnEventView.html), not just individual events (note that `EventTrigger` implements `From<Event>`, which is why any event can be used directly).

You can also use any [`F: Fn(&Event) -> bool`](https://docs.rs/cursive/latest/cursive/event/struct.EventTrigger.html#impl-From%3CF%3E-for-EventTrigger). This way you can have more control, for example catching only some kind of events (for example only `Char(_)` events).

**Reply by smallB007** [reply]
Hi,
Thanks for the info. I am afraid more help is needed. I've tried to follow your advice and do:
`let convert = |e: &cursive::event::Event| true;`
But I don't know how to check if e was a Char(_).
Could you please help me with that?
Thanks
[EDIT]
I've managed to do it. I did it in the following way (could you please suggest better, more idiomatic way, or is that OK?):
```
    let convert = |ev: &cursive::event::Event| {
        match ev {
            Event::Char(_) => {
                eprintln!("char")
            }
            _ => {
                eprintln!("Not a char")
            }
        }

        true
    };
```
Thanks

**Reply by smallB007** [reply]
Another question, that is really follow up on the previous one. How do I access the information, which char was pressed in the on_event handler? For the moment I have it done this way:

```
.on_event(convert, |s| {
//How to get the information which char was pressed inside this on_event handler?
            eprintln!("RETURNED TRUE");

        });
```
Thanks

**Reply by gyscos** [reply]
* You can use the [`matches!`](https://doc.rust-lang.org/std/macro.matches.html) macro to easily check that: `matches!(ev, Event::Char(_))` will return `true` if it's that type. But a regular `match` block is pretty idiomatic too, some people dislike macros.
* You can use [`OnEventView::on_event_inner()`](https://docs.rs/cursive/latest/cursive/views/struct.OnEventView.html#method.on_event_inner). It takes a slightly more complex closure that returns a `Option<EventResult>`. Return `None` here to ignore the event, and `Some(EventResult::with_cb(|s| { ... }))` to include a callback similar to what you would give in `on_event()`.

**Reply by smallB007** [reply]
Hi, that indeed helped. Thank you for your help. I really appreciate it!
